### PR TITLE
fix: Add Markdown support for showInputBox

### LINKED ISSUE #1920: Add API for text input dialog (eg. as used for Pull-secret input prompt)
### Is your enhancement related to a problem? Please describe

In case of OpenShift Local we need to ask user to provide pullsecret, which is a bug string, now we use `showInputBox` function which is not convenient for entering long string data.

### Describe the solution you'd like

It would be nice if podman-desktop provides an API for extension to create such dialog.
API should provide a way to configure title, has ability to validate input before confirmation button, and ability to provide textual description of required data, with links to external documentation when user could read more.

Example of UI:

Mockup from https://github.com/containers/podman-desktop-internal/issues/84

### Describe alternatives you've considered

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by jeffmaury**
Fixes #1920

### What does this PR do?

Enhance showInputBox with multiLine and markdownDescription

### Screenshot/screencast of this PR

image

Muliline:

image

### What issues does this PR fix or reference?

Fixes #1920

### How to test this PR?

Add the following code in _packages/main/src/plugin/index.ts_ line 350:

```
    statusBarRegistry.setEntry('area', false, 0, undefined, 'Area', 'fa fa-bell', true, 'show-area', undefined);
    commandRegistry.registerCommand('show-area', async () => {
      await inputQuickPickRegistry.showInputBox({
        prompt: 'Enter the pull secret',
        markdownDescription: 'To put container images from the regsitry, a *pull secret* is necessary. You can get a pull secret from the [Red Hat OpenShift Local download page](https://developers.redhat.com/products/openshift-local/overview). Use the *"Copy pull secret"* option and paste the content into the field above',
        multiline: true,
      });
    });
```

and click on the first bell in the status area

### CODE REVIEW

File: packages/renderer/src/lib/dialogs/QuickPickInput.svelte [hunk 1]
```diff
@@ -284,7 +299,7 @@ function handleKeydown(e: KeyboardEvent) {
           {#if validationError}
             <div class="text-gray-400 border border-red-700 relative w-full bg-red-700 px-1">{validationError}</div>
           {:else}
-            <div class="relative text-gray-400 pt-2 px-1 h-6 overflow-y-auto">{prompt}</div>
+            <div class="relative text-gray-400 pt-2 px-1 h-6 overflow-y-auto"><Markdown>{prompt}</Markdown></div>
```
Comment (benoitf) [inline packages/renderer/src/lib/dialogs/QuickPickInput.svelte]: it needs to be an optional field like `markdownDescription`

Comment (benoitf) [reply packages/renderer/src/lib/dialogs/QuickPickInput.svelte]: or markdownPrompt but it looks like it's more a description that we wants here to add extra details

**Comment by benoitf** [comment]
@jeffmaury do you have a screnshot on how it's rendering to add in the body of the PR ?

**Review by benoitf — APPROVED** [review]
It looks the rendering 
image

is not exactly complete compared to the mockup
image

- text too big for the description
- missing horizontal purple bar between title and description
- missing the info icon
- links are blue and not in purple
- textarea should have resize-none

but it could be iterated

**Comment by jeffmaury** [comment]
> It looks the rendering image
> 
> is not exactly complete compared to the mockup image
> 
>     * text too big for the description
> 
>     * missing horizontal purple bar between title and description
> 
>     * missing the info icon
> 
>     * links are blue and not in purple
> 
>     * textarea should have resize-none
> 
> 
> but it could be iterated

Yes but I discussed with @deboer-tim and we agreed to minimize changes in favor of stability

**Comment by gbraad** [comment]
Will we over time get the dialog as discussed? Or is this a final decision?
/cc: @mairin

**Comment by benoitf** [comment]
intermediate step

**Pull request merged**